python/amici/sbml_import.py

```diff
@@ -1188,41 +1192,53 @@ def _check_unsupported_functions(sym, expression_type, full_sym=None):
             _check_unsupported_functions(fun, expression_type)
 
 
-def _parse_special_functions(sym, toplevel=True):
+def _parse_special_functions(sym):
     """Recursively checks the symbolic expression for functions which have be
     to parsed in a special way, such as piecewise functions
 
         Arguments:
             sym: symbolic expressions @type sympy.Basic
-            toplevel: as this is called recursively,
-                are we in the top level expression?
+
         Returns:
 
         Raises:
     """
-    args = tuple(_parse_special_functions(arg, False) for arg in sym._args)
+    args = tuple(_parse_special_functions(arg) for arg in sym._args)
 
+    # Do we have piecewise expressions?
     if sym.__class__.__name__ == 'abs':
         return sp.Abs(sym._args[0])
-    elif sym.__class__.__name__ == 'xor':
```

Comment:
... here

Reply:
I'd propose to just add the exception line in a new commit on (an up-to-date) `test_action`. That will be the least painfull :)